Packs/Tenable_io/Integrations/Tenable_io/Tenable_io.py

```diff
@@ -1114,33 +1248,234 @@ def export_vulnerabilities_command(args: Dict[str, Any]) -> PollResult:
         return request_uuid_export_vulnerabilities(args)
 
 
+def scan_filters_human_readable(filters: list) -> str:
+    context_to_hr = {
+        'name': 'Filter name',
+        'readable_name': 'Filter Readable name',
+        'type': 'Filter Control type',
+        'regex': 'Filter regex',
+        'readable_regex': 'Readable regex',
+        'operators': 'Filter operators',
+        'group_name': 'Filter group name',
+    }
+    return tableToMarkdown(
+        'Tenable IO Scan Filters',
+        [d | d.get('control', {}) for d in filters],
+        headers=list(context_to_hr),
+        headerTransform=context_to_hr.get,
+        removeNull=True)
+
+
+def list_scan_filters_command(client: Client) -> CommandResults:
+
+    response_dict = client.list_scan_filters()
+    filters = response_dict.get('filters', [])
+
+    return CommandResults(
+        outputs_prefix='TenableIO.ScanFilter',
+        outputs_key_field='name',
+        outputs=filters,
+        readable_output=scan_filters_human_readable(filters),
+        raw_response=response_dict)
+
+
+def scan_history_readable(history: list) -> str:
+    context_to_hr = {
+        'id': 'History id',
+        'scan_uuid': 'History uuid',
+        'status': 'Status',
+        'is_archived': 'Is archived',
+        'custom': 'Targets custom',
+        'default': 'Targets default',
+        'visibility': 'Visibility',
+        'time_start': 'Time start',
+        'time_end': 'Time end',
+    }
+    return tableToMarkdown(
+        'Tenable IO Scan History',
+        [d | d.get('targets', {}) for d in history],
+        headers=list(context_to_hr),
+        headerTransform=context_to_hr.get,
+        removeNull=True)
+
+
+def scan_history_params(args: dict) -> dict:
+    return {
+        'sort': ','.join(
+            f'{field}:{order}'
+            for field, order
+            in zip(
+                argToList(args.get('sortFields')),
+                # * 3 makes it apply to all sortFields even if it has only one value
+                argToList(args.get('sortOrder')) * 3)),
```

Comment:
what about the case where we have 3 sort fields and two orders provided? This might not be what the customer expected.